src/main/java/org/tikv/common/util/ClientUtils.java

```diff
@@ -121,7 +121,7 @@ public static void appendBatches(
 
   public static Map<TiRegion, List<ByteString>> groupKeysByRegion(
       RegionManager regionManager, List<ByteString> keys, BackOffer backoffer) {
-    return groupKeysByRegion(regionManager, new ArrayList<>(keys), backoffer, false);
+    return groupKeysByRegion(regionManager, keys, backoffer, false);
```

Comment:
`keys` is sorted in line 137.
copy is necessary here to avoid multi-threads modification to the same keys List.

Reply:
in which condition do we need to call groupKeysByRegion in concurrently?

Reply:
ok, let's remove it.
line 119 also do copy.

Reply:
great!